devito/operator/operator.py

```diff
@@ -871,11 +872,14 @@ def apply(self, **kwargs):
         # Post-process runtime arguments
         self._postprocess_arguments(args, **kwargs)
 
-        # Output summary of performance achieved
-        return self._emit_apply_profiling(args)
+        # In case MPI is used restrict result logging to one rank only
+        if configuration['mpi']:
+            with mpi_switch_log(log_level='DEBUG', comm=args.comm):
```

Comment:
why make the "switch log" mpi-specific?

Reply:
I wanted to avoid confusion between having sth like set_log_level and switch_log
I am happy if you propose a name for that

Reply:
`switch_log_level` is fine

Reply:
done